addons/knobs/src/index.js

```diff
@@ -8,12 +8,12 @@ export function knob(name, options) {
   return manager.knob(name, options);
 }
 
-export function text(name, value, groupId) {
-  return manager.knob(name, { type: 'text', value, groupId });
+export function text(name, value, groupId, { onChange } = {}) {
+  return manager.knob(name, { type: 'text', value, groupId, onChange });
 }
 
-export function boolean(name, value, groupId) {
-  return manager.knob(name, { type: 'boolean', value, groupId });
+export function boolean(name, value, groupId, { onChange } = {}) {
+  return manager.knob(name, { type: 'boolean', value, groupId, onChange });
 }
 
 export function number(name, value, options = {}, groupId) {
```

Comment:
Why the number know is different?

Reply:
Because it had `options` before introducing `groupId`.

I think we should move `groupId` inside options and deprecate the old API. @JalilArfaoui please let me know if you need any help with that

Reply:
(Sorry for the big delay in my response). 

Wait, @Hypnosphi ... Are you saying you want to change all knobs API to this: 

```js

export function text(name, value, { groupId, onChange } = {}) 
export function boolean(name, value, { groupId, onChange } = {}) 
export function number(name, value, {groupId, onChange} = {} ) 
export function color(name, value, { groupId, onChange } = {}) 
export function object(name, value,{  groupId, onChange } = {}) 

// --> those ones have more changes
export function select(name, value, { groupId, onChange } = {})
export function array(name, value, {  separator = ',', groupId, onChange } = {})
export function date(name, { value = new Date(), groupId, onChange } = {})
export function button(name,{  callback, groupId, onChange } = {}) 
export function files(name, value = [], { accept, onChange } = {})
```

??

It's easy to do ... but do we all agree that this will be a major breaking change in addon-knobs API ? 

(My first proposal was retro-compatible)

Should I go ahead with those definitions @Keraito @danielduan @Hypnosphi

Reply:
I've just seen that this PR was marked with milestone 4.0.0 and that we are already in 4.0.0 release candidate ? Has the API been frozen ? Can this still make it to 4.0.0 if I push the modification very soon ?

Reply:
I think it's too late already for breaking changes =( But we can make it backward compatible.

for all knobs (besides number):

```js
function backwardCompatOptions(args) {
  const [name, value, arg1, arg2] = args;

  if (arg1 !== null && arg1 !== 'object') { 
    return {
      groupId: arg1,
      ...(arg2 | {}),
    }
  }
  
  return arg1 || {};
}

export function boolean(name, value) {
  const options = backwardCompatOptions(arguments);
  const = { groupId, onChange } = options;

  // ...
}
```

for the `number` should be something similar

Reply:
You're so right ! I'll do it like that, so that we can support both new and (deprecated) legacy API.